### PR TITLE
blacklist pattern for alarm names

### PR DESCRIPTION
**Pull request by zmaleki**
added a blacklist parameter where alarm names matching that pattern would not generate metrics

### CODE REVIEW

File: template.yml [hunk 1]
```diff
@@ -10,11 +10,17 @@ Metadata:
...
-    Labels: ["metrics", "mttr", "mtbf"]
+    Labels: ['metrics', 'mttr', 'mtbf']
     HomePageUrl: https://github.com/MechanicalRock/aws-ops-metrics
-    SemanticVersion: 1.3.3
+    SemanticVersion: 1.3.4
     SourceCodeUrl: https://github.com/MechanicalRock/aws-ops-metrics
 
+Parameters:
+  AlarmNameBlackListPattern:
+    Description: 'a regex pattern which would be matched agains alarm name in order to blacklist it from generating health metrics'
```
Comment (temyers) [inline template.yml]: Add a comment about the default regex.

File: template.yml [hunk 2]
```diff
@@ -34,70 +40,79 @@ Resources:
...
-    Type: "AWS::Serverless::Function"
+    Type: 'AWS::Serverless::Function'
     Properties:
       Description: Create CloudWatch mttr metric from cloudwatch alarm events
       Handler: mttr.handler
       Runtime: nodejs12.x
       CodeUri: .build/src
+      Environment:
+        Variables:
+          IGNORED_ALARM_NAME_PATTERN: !Ref AlarmNameBlackListPattern
```
Comment (temyers) [inline template.yml]: can we change to `ALARM_NAME_BLACKLIST_PATTERN` - makes it consistent with language elsewhere

File: test/common.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      shouldMatch: false,
+    },
+    {
+      givenData:
+        'TargetTracking-table/ci-crawl-status-check-ProvisionedCapacityHigh-36d74b4f-f037-40e0-9207-27a9e86ddc6',
+      shouldMatch: false,
+    },
+    {
+      givenData:
+        'TargetTracking-table/ci-crawl-status-check-ProvisionedCapacityHigh-36d74b4f-f037-40e0-gggg-9207-27a9e86ddc6',
```
Comment (temyers) [inline test/common.test.ts]: why doesn't this one match?

Might be useful to add a `description` to each example - even if it's not used (or add it to the test name)

Comment (zmaleki) [reply test/common.test.ts]: Because it's got gggg in the guid. guid pattern are hex characters

File: template.yml (hunk 1 above)
Comment (zmaleki) [reply template.yml]: Done

File: template.yml (hunk 2 above)
Comment (zmaleki) [reply template.yml]: Done

File: test/common.test.ts (hunk 3 above)
Comment (zmaleki) [reply test/common.test.ts]: Done

File: template.yml [hunk 4]
```diff
@@ -19,6 +19,7 @@ Parameters:
   AlarmNameBlackListPattern:
     Description: 'a regex pattern which would be matched agains alarm name in order to blacklist it from generating health metrics'
     Type: String
+    # Below default regex pattern matches the auto generated alarm names created by dynamodb's auto scaling
```
Comment (temyers) [inline template.yml]: Can you put this in the description.

File: test/common.test.ts [hunk 5]
```diff
@@ -3,50 +3,64 @@ import { matchesBlackList } from '../src/common';
...
       shouldMatch: false,
+      description:
+        'Should include any of these in the name : -AlarmHigh|-AlarmLow|-ProvisionedCapacityHigh|-ProvisionedCapacityLow',
     },
     {
       givenData:
-        'TargetTracking-table/ci-crawl-status-check-ProvisionedCapacityHigh-36d74b4f-f037-40e0-9207-27a9e86ddc6',
+        'TargetTracking-table/ci-crawl-status-check-ProvisionedCapacityHigh-36d74b4f-f037-40e0-9207-27a9e98ddc6',
       shouldMatch: false,
+      description: 'Guid has a missing character',
```
Comment (temyers) [inline test/common.test.ts]: :+1:

**Pull request merged**